Fixed #24728 -- Rename mime_type to content_type for syndication feeds

raphaelm opened the pull request:
Renamed the mime_type properties of RssFeed and Atom1Feed to
content_type and start deprecation for the old names.

raphaelm commented:
buildbot, test this please.

timgraham commented:
This wouldn't work if someone has subclassed the fields and set their own `mime_type` -- this doesn't seem like it would be common though, so maybe we can ignore that. We need to update the places that use the deprecated attribute, e.g. `item['enclosure'].mime_type`. Also a mention in docs/internals/ddeprecation.txt is missing.

timgraham commented:
No need for the "buildbot, test this please." comment -- builds will run automatically and that comment triggers extra builds.

raphaelm commented:
Okay, I got that wrong about buildbot, thanks.

Concerning ``item['enclosure'].mime_type``, I believe it is correct this way. The scope of the ticket was to remove the term "mime type" form everything related to HTTP (such as: the rendered feed), in order to match the terminology of the HTTP header ``Content-Type``. 

However, ``item['enclosure'].mime_type`` refers to the type of an ``Enclosure`` object. In this case, "mime type" is the commonly used term (see for example [rssboard.org](http://www.rssboard.org/rss-profile#element-channel-item-enclosure)). Therefore, I did not deprecate this. (Although we could do, if we want, for consistency.)

Django itself never uses the ``mime_type`` attribute of ``Atom1Feed`` or ``RssFeed`` outside of tests, neither is it a documented attribute. Additionally, the attribute does not exist on the base ``SyndicationFeed`` class. Therefore, to my understanding it wasn't even a public API, so I think we can indeed ignore the subclassing issue.

raphaelm commented:
Please ignore my last comment, I understood something wrong there. I will look into it.

raphaelm commented:
I missed one usage when grepping. Fixed it now, complete test suite passes. 

Also, I added it to the deprecation list. For enclosures, see my comment above. Thank you for the feedback! :)

ubernostrum commented:
This looks good to me, waiting to see if tests pass.

timgraham commented:
merged in 5c125f63f77e169a459c11bab0f639c83fabc6e2, thanks!
